### DISCUSSION TITLE
How to initialise VectorOfPostprocessors

### DISCUSSION THREAD

**Discussion by fervaldies** [idaholab/moose#22854] [Q&A General] [answered]
Hello, I am currently using a "getPostprocessorValue" and "getPostprocessorValueOld" to receive the temperature of one location in the 3D model. What I want to do is to change the position of this Postprocessor at a certain time step in the simulation, and I was thinking of using a vector of postprocessors and then take the, for example, temp[1] and then temp[2] for different time steps to change the location of the point to collect the temperature. Is there any getVectorPostprocessorValue or something similar? If so, how do I initialise it in the .C and .h files? The .i code would look something like:

```
[Materials]
 [./vol_h]
    temp_passes = 'temp_pass1 temp_pass2 temp_pass3'
 []
[]

...
...

[Postprocessors]
  [./temp_pass1]
    type = PointValue
    variable = temp
    point = '2 2 2'
  [../]
  [./temp_pass2]
    type = PointValue
    variable = temp
    point = '2 3 1'
  [../]
  [./temp_pass3]
    type = PointValue
    variable = temp
    point = '1 2 3'
  [../]
[]
```
Being temp_pass1, temp_pass2, and temp_pass3 the postprocessors that I would like to select depending on the time step. So something like temp_passes[1] or temp_passes[2], as an example.

Thank you in advance

**Comment by GiudGiud** [comment, marked as answer]
Hello

There are a few options here:
- use the VPP interface. Then there is a [getVectorPostprocessorValue](https://mooseframework.inl.gov/docs/doxygen/moose/classVectorPostprocessorInterface.html#a1fc43efa0071129d68bc0342ad45cea0) routine. There are similar routines for "old" values.
You may need to add it to the list of classes your object inherits from if it's not already (99% chance it is)
https://mooseframework.inl.gov/docs/doxygen/moose/classVectorPostprocessorInterface.html

- use the Reporter interface and use the getReporterValue with the name of the VPP and the name of the vector within the VPP. 
This is because all VPPs are reporters
https://mooseframework.inl.gov/docs/doxygen/moose/classReporterInterface.html

- dont use your strategy by changing the indexing, but by changing the position in the postprocessor and, if necessary, performing the geometric search again for the new position.
Changing the position can be done by making it controllable and using the Controls system.
https://mooseframework.inl.gov/syntax/Controls/index.html

Guillaume
